core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/AbstractCamelModelElement.java

```diff
@@ -438,7 +440,12 @@ public AbstractCamelModelElement getInputElement() {
 	}
 
 	public boolean isEndpointElement() {
-		return getParameter("uri") != null && ((String) getParameter("uri")).trim().length() > 0;
+		if(Arrays.asList(ENDPOINT_TYPE_FROM, ENDPOINT_TYPE_TO, ENDPOINT_NODE_NAME).contains(getNodeTypeId())){
+			Object uriParameter = getParameter(URI_PARAMETER_KEY);
+			return uriParameter != null && ((String) uriParameter).trim().length() > 0;
```

Comment:
maybe, I kept this part of the old definition. I'm not sure about edge case that it can imply to remove it.